pkg/volume/rbd/rbd_util.go

```diff
@@ -258,36 +278,48 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) (string, error) {
 
 	devicePath, found := waitForPath(b.Pool, b.Image, 1)
 	if !found {
+		getAttachMutex.LockKey(globalPDPath)
+		defer getAttachMutex.UnlockKey(globalPDPath)
+
+		fmt.Println("<><><><><> AttachDisk devicePath is not found")
+
 		_, err = b.exec.Run("modprobe", "rbd")
 		if err != nil {
 			glog.Warningf("rbd: failed to load rbd kernel module:%v", err)
 		}
 
 		// Currently, we don't acquire advisory lock on image, but for backward
 		// compatibility, we need to check if the image is being used by nodes running old kubelet.
-		found, rbdOutput, err := util.rbdStatus(&b)
-		if err != nil {
-			return "", fmt.Errorf("error: %v, rbd output: %v", err, rbdOutput)
-		}
-		if found {
-			glog.Infof("rbd image %s/%s is still being used ", b.Pool, b.Image)
-			return "", fmt.Errorf("rbd image %s/%s is still being used. rbd output: %s", b.Pool, b.Image, rbdOutput)
-		}
+		//		found, rbdOutput, err := util.rbdStatus(&b)
```

Comment:
add a comment here why dropping rbd status. 

rbd watcher may not be cleaned up even after reboot, per http://lists.ceph.com/pipermail/ceph-users-ceph.com/2016-August/012140.html

cc @dillaman

Reply:
Watches are not cleaned up due to a "reboot" -- but they are cleaned up if they expire (i.e. >30 seconds w/o a heartbeat from the client per default settings).

Reply:
thanks @dillaman 
in this case, waiting for 30s and retry watcher is more reliable

Reply:
@dillaman @rootfs  it must be OSD bug as in my cases Watcher stays on all the time and the image is not mapped.

sbezverk@kube-4:~/developing/go/src/k8s.io/kubernetes$ date; sudo rbd status raw-volume --pool kubernetes
Wed Dec 20 10:04:19 EST 2017
Watchers:
        watcher=192.168.80.235:0/3789045165 client.64439 cookie=1
sbezverk@kube-4:~/developing/go/src/k8s.io/kubernetes$ date; sudo rbd status raw-volume --pool kubernetes
Wed Dec 20 10:04:51 EST 2017
Watchers:
        watcher=192.168.80.235:0/3789045165 client.64439 cookie=1
sbezverk@kube-4:~/developing/go/src/k8s.io/kubernetes$ date; sudo rbd status raw-volume --pool kubernetes
Wed Dec 20 10:05:14 EST 2017
Watchers:
        watcher=192.168.80.235:0/3789045165 client.64439 cookie=1

Do we really need to keep rbdStatus? which kubelet version still requires this check?

Reply:
rbdStatus is to be compatible with kubelets prior to 1.9 that use advisory locks. The idea is to stop using the rbd image if someone else already uses it to enforce RWO.